fitbenchmarking/results_processing/plots.py

```diff
@@ -263,58 +180,73 @@ def plot_summary(cls, categories, title, options, figures_dir):
         else:
-            ax.plot(first_result.data_x,
-                    first_result.data_y,
-                    **cls.data_plot_options)
-
-        # Setup x for rest of plots
-        x = first_result.data_x
-        x = first_result.data_x[first_result.sorted_index]
-
-        for (key, results), colour in zip(categories.items(), colours):
+            error_y = None
+        plotlyfig.add_trace(go.Scatter(x=first_result.data_x,
+                                       y=first_result.data_y,
+                                       error_y=error_y,
+                                       mode='markers',
+                                       name='Data',
+                                       marker=cls.data_marker))
+
+        for (key, results), colour in zip(categories.items(), plotly_colours):
             # Plot category
             for result in results:
                 if result.params is not None:
-                    y = result.fin_y[result.sorted_index]
-                    plot_options = cls.summary_best_plot_options \
-                        if result.is_best_fit else cls.summary_plot_options
-                    plot_options['color'] = colour
-                    plot_options['label'] = key if result.is_best_fit else ''
-
-                    ax.plot(x, y, **plot_options)
-                    # log scale
-                    if result.plot_scale == "loglog":
-                        ax.set_xscale("log", nonpositive='clip')
-                        ax.set_yscale("log", nonpositive='clip')
-                    elif result.plot_scale == "logy":
-                        ax.set_yscale("log", nonpositive='clip')
-                    elif result.plot_scale == "logx":
-                        ax.set_xscale("log", nonpositive='clip')
-                    ax.set_xlabel("X")
-                    ax.set_ylabel("Y")
-                    ax.set_title(title,
-                                 fontsize=10)
-                    ax.legend(loc="upper left")
-                    fig.set_tight_layout(True)
-
-        fname = f'summary_plot_for_{first_result.sanitised_name}.png'
-        fig.savefig(os.path.join(figures_dir, fname))
-        plt.close(fig)
-        return fname
+                    line = cls.summary_best_plot_line \
+                        if result.is_best_fit else cls.summary_plot_line
+
+                    line["color"] = colour
+                    label = key if result.is_best_fit else ''
+                    if result.is_best_fit:
+                        line = cls.summary_best_plot_line
+                        line["color"] = colour
+                    else:
+                        line = cls.summary_plot_line
+                        transparency = 0.5
+                        line["color"] = 'rgba' + colour[3:-1] + ', ' + \
+                            str(transparency) + ')'
+
+                    plotlyfig.add_trace(go.Scatter(
+                        x=first_result.data_x[first_result.sorted_index],
```

Comment:
If we're not setting x up before the loop I think it's more readable to just use result
```suggestion
                        x=result.data_x[result.sorted_index],
```